framework/di/Container.php

```diff
@@ -364,6 +365,9 @@ protected function build($class, $params, $config)
         }
 
         $dependencies = $this->resolveDependencies($dependencies, $reflection);
+        if (!$reflection->isInstantiable()) {
+            throw new NotInstantiableException($reflection->name);
```

Comment:
I cann't see much use of `NotInstantiableException`. Better use just `InvalidConfigException` or `InvalidCallException`.
Inside the framework we do not use sophisticated exception hierrarchy.